### PR TITLE
Remove restriction preventing to have | surrounded by whistespace characters

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes https://github.com/rinja-rs/rinja/issues/245.

Just checked and jinja2 works when filter operators are surrounded with whitespace characters. From this point, there doesn't seem to be any reason for `rinja` to not allow it as well.

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
The book was still wrong about why whitespace characters were needed. ^^'

File: rinja_parser/src/lib.rs [hunk 1]
```diff
@@ -899,20 +899,10 @@ fn filter<'a>(
     i: &'a str,
     level: &mut Level,
 ) -> ParseResult<'a, (&'a str, Option<Vec<WithSpan<'a, Expr<'a>>>>)> {
-    let (j, _) = skip_ws0.parse_next(i)?;
-    let had_spaces = i.len() != j.len();
-    let (j, _) = ('|', not('|')).parse_next(j)?;
+    let (j, _) = (ws('|'), not('|')).parse_next(i)?;
```
Comment (Kijewski) [inline rinja_parser/src/lib.rs:902]:
```suggestion
    let (j, _) = ws(('|', not('|'))).parse_next(i)?;
```

Comment (GuillaumeGomez) [reply rinja_parser/src/lib.rs:902]: 👍

**Review by Kijewski — APPROVED** [review]
Yeah, it's easier to read with spaces around the `|`.

**Comment by GuillaumeGomez** [comment]
Agreed. :)

**Pull request merged**